src/sync/grandpa_warp_sync.rs

```diff
@@ -338,7 +360,7 @@ impl<TSrc> StorageGet<TSrc> {
     pub fn inject_value(
         self,
         value: Option<impl Iterator<Item = impl AsRef<[u8]>>>,
-    ) -> GrandpaWarpSync<TSrc> {
+    ) -> (GrandpaWarpSync<TSrc>, Option<(Error, HostVmPrototype)>) {
```

Comment:
```suggestion
    ) -> (GrandpaWarpSync<TSrc>, Option<Error>) {
```

You don't need to pass back the virtual machine, since the error is that the runtime in the virtual machine has a problem.

Reply:
Oh right, so the comment `// TODO: we use 'Oneshot' because the VM is thrown away afterwards; ideally it wouldn't be be thrown away` is referring to what happens to the VM upon success?

Reply:
Yes!